src/dodal/devices/zocalo/zocalo_results.py

```diff
@@ -237,9 +242,18 @@ def _receive_result(
             self.transport.ack(header)  # type: ignore # we create transport here
 
             results = message.get("results", [])
-            self._raw_results_received.put(
-                {"results": results, "ispyb_ids": recipe_parameters}
-            )
+
+            if self.use_fastest_zocalo_result:
+                self._raw_results_received.put(
+                    {"results": results, "ispyb_ids": recipe_parameters}
+                )
+            else:
+                #Only add to queue if results are from CPU
+                if recipe_parameters.get('gpu') == None:
```

Comment:
Also here
```suggestion
                if recipe_parameters.get('gpu') is None:
```

Reply:
In fact again truthiness of None is probably better
```suggestion
                if recipe_parameters.get('gpu'):
```

Reply:
Yeah, I always prefer to explicitly compare it to None if it can be True/False or None